scripts/irods/test/test_misc.py

```diff
@@ -317,3 +319,29 @@ def test_server_correctly_cleans_up_proc_files__issue_6231(self):
         # Show that the proc directory will eventually become empty.
         lib.delayAssert(lambda: len(os.listdir(paths.proc_directory())) == 0)
 
+    def test_iinit_does_not_crash_from_long_home_directory__issue_5411(self):
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            # Create new home directory for test
+            temp_home = os.path.join(temp_dir_name, 'crazy', 'long', 'temp', 'home', 'directory',
+                                     'like', 'this', 'is', 'longer', 'than', 'the', 'solos', 'in', 'chameleon')
+            os.makedirs(temp_home)
+
+            # Setup environment
+            temp_env = os.environ.copy()
+            temp_env['HOME'] = temp_home
+
+            # Execute iinit
+            user_input = os.linesep.join(
+                ['localhost', '1247', 'rods', 'tempZone', 'rods']) + os.linesep
+            res = subprocess.run('iinit', input=user_input.encode(
+            ), stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=temp_env)
+
+            self.assertEqual(res.returncode, 0)
```

Comment:
Should 0 be returned if the size of the home collection is too large?

Reply:
I think we would get an error in `obf.cpp` if we put in a very long home directory of ~1000+ chars.

In this case, the string was no more than 200 chars, very likely less, so we expect `iinit` to work fine.

Reply:
Did the test fail with 200 characters before?
What happens if the path exceeds 1k characters?

Reply:
Originally, the program failed on an input of 79 chars.

I'll throw some longer inputs at it to see what happens. 👍

Reply:
Running with a HOME greater than 1098 characters, including `/`s, leads to a return of 1 from main.
The following error is also printed: `ERROR: Save Password failure status = -900000 FILE_OPEN_ERR`.

Reply:
That error message doesn't look very helpful.

Do we know what caused the failure? If so, we should be more explicit if possible.

Reply:
My guess would be `obfSavePw` failing to write the `.irodsA` file? But that would be strange since I would have expected `iinit` to fail before reaching that point. Maybe the directory is created successfully, but adding `.irodsA` to the end to save the obfuscated password fails because only then is the path too long.

Reply:
Updated so when the path to `.irodsA` is too long, `USER_INPUT_PATH_ERR` is returned. I'm curious if there's a more appropriate error status available for this case?

Reply:
What would you suggest the name be for a new error code?
What would the generic name be (so we can use it in other situations)?

Reply:
I found `USER_PATH_EXCEEDS_MAX`, if that seems better.

Reply:
Just going to change `ERROR: Save Password failure` to `ERROR: Failed to save password!` in the other PR. Maybe the `!` will help separate the `status` a bit better, too.